### PR TITLE
update outdated dependencies in helm chart

### PR DESCRIPTION
**Pull request by mikkeschiren**

### CODE REVIEW

**Comment by mikkeschiren** [comment]
Bitnami helm chart now using OCI repo, this pull request updates the repo and also the version of the charts, the version added has been tested together with Zulip.

File: kubernetes/chart/zulip/templates/_helpers.tpl [hunk 1]
```diff
@@ -65,16 +65,17 @@ Create the name of the service account to use
...
 - name: DB_HOST_PORT
-  value: "{{ template "postgresql.service.port" .Subcharts.postgresql }}"
+  value: "{{ template "postgresql.v1.service.port" .Subcharts.postgresql }}"
 - name: DB_USER
   value: "postgres"
 - name: SETTING_MEMCACHED_LOCATION
   value: "{{ template "common.names.fullname" .Subcharts.memcached }}:11211"
 - name: SETTING_RABBITMQ_HOST
-  value: "{{ template "rabbitmq.fullname" .Subcharts.rabbitmq }}"
+  value: "{{ template "common.names.fullname" .Subcharts.rabbitmq }}"
```
Comment (timabbott) [inline kubernetes/chart/zulip/templates/_helpers.tpl:78]: Can you point to documentation or some other way to check that these changes to the template are correct?

Comment (mikkeschiren) [reply kubernetes/chart/zulip/templates/_helpers.tpl:78]: i am not really shure when this was changed - but as you can see from fullname from memcached, the change matches to the memcached value for fullname, you can also see this in the _helpers.tpl in the bitnami repo: https://github.com/bitnami/charts/blob/main/bitnami/rabbitmq/templates/_helpers.tpl#L34

Comment (mikkeschiren) [reply kubernetes/chart/zulip/templates/_helpers.tpl:78]: WIth the suggested change `SETTING_RABBITMQ_HOST` get the value of `zulip-rabbitmq` (with a helm install name set to zulip), and the service name `zulip-rabbitmq`

**Comment by timabbott** [comment]
Merged, after tweaking the commit message to use a `helm:` prefix. 
Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.

File: kubernetes/chart/zulip/Chart.yaml [hunk 2]
```diff
@@ -15,26 +15,26 @@ version: 0.9.20
...
-    version: 16.8.7
+    version: 20.1.4
   - name: postgresql
-    repository: https://charts.bitnami.com/bitnami
+    repository: oci://registry-1.docker.io/bitnamicharts
     tags:
       - postgresql
     # Note: values.yaml overwrites posgresql image to zulip/zulip-postgresql:14
-    version: 11.1.22
+    version: 15.5.32
```
Comment (alexmv) [inline kubernetes/chart/zulip/Chart.yaml:37]: I'm not sure if this change (upgrading across several major PostgreSQL versions) means that folks using the Helm chart may lose their data, or require a manual dump-and-reload process.  I assume you tested across this change, and it Just Worked to upgrade and preserve the backing data in the PostgreSQL server image?

**Pull request merged**